File: AR7ExportToolbar/ToolStripExport/Exporters/ExcelSectionReportExporter.cs
using System;
using System.Collections.Generic;
using System.Collections.Specialized;
using System.Linq;
using System.Text;
using GrapeCity.ActiveReports.Document;
using GrapeCity.ActiveReports.Export;

namespace AR7ExportToolbar.ToolStripExport.Exporters
{
	public class ExcelSectionReportExporter : SectionReportExporter
	{
		private const string TypeName = "GrapeCity.ActiveReports.Export.Excel.Section.XlsExport";
		private const string AssemblyName = "GrapeCity.ActiveReports.Export.Excel.v7";

		public ExcelSectionReportExporter()
			: base(TypeName, AssemblyName)
		{
		}

		public override void Export(SectionDocument document, string filename, NameValueCollection settings)
		{
			throw new NotImplementedException();
		}

		public override string DefaultExtension
		{
			get { return "xls"; }
		}

		public override string FileDialogFilter
		{
			get { return "Excel Files (*.xls)|*.xls|All Files (*.*)|*.*"; }
		}
	}
}

File: AR7ExportToolbarVB/ToolStripExport/Exporters/ExcelSectionReportExporter.vb
Imports GrapeCity.ActiveReports.Document
Imports System.Collections.Specialized

Public Class ExcelSectionReportExporter
	Inherits SectionReportExporter
	Private Const TypeName As String = "GrapeCity.ActiveReports.Export.Excel.Section.XlsExport"
	Private Const AssemblyName As String = "GrapeCity.ActiveReports.Export.Excel.v7"

	Public Sub New()
		MyBase.New(TypeName, AssemblyName)
	End Sub

	Public Overrides Sub Export(document As SectionDocument, filename As String, settings As NameValueCollection)
		Throw New NotImplementedException()
	End Sub

	Public Overrides ReadOnly Property DefaultExtension() As String
		Get
			Return "xls"
		End Get
	End Property

	Public Overrides ReadOnly Property FileDialogFilter() As String
		Get
			Return "Excel Files (*.xls)|*.xls|All Files (*.*)|*.*"
		End Get
	End Property
End Class
